### ISSUE TITLE
normalizeCachePath() can only ever return default, APP_CONFIG_CACHE ignored

### ISSUE THREAD

**Issue by leonardchallis** [laravel/framework#46630]
- Laravel Version: 10.5.0
- PHP Version: 8.2.4
- Database Driver & Version: MySQL 8.0.32

### Description:

From my understanding, first the `LoadEnvironmentVariables` bootstrapper runs, followed by `LoadConfiguration`. Inside the latter, `normalizeCachePath` is called from `getCachedConfigPath()` and it checks for values in `Env`, but there are none - and `$_ENV` is empty also. Despite the first bootstrapper, it seems as if they still aren't registered in the environment at this point. I thought maybe it was because `afterLoadingEnvironment` had not been called but it seems [it isn't supposed to be](https://github.com/laravel/framework/pull/27044#issuecomment-451365799).

This means the env is always null and the default is always returned.

### Steps To Reproduce:

- Install Laravel
- Set `APP_CONFIG_CACHE`
- Notice it's ignored / `dd($_ENV)` or any other env var from inside `normalizeCachePath()`

Related: https://github.com/laravel/framework/issues/32357

## Update

I realised that, `APP_CONFIG_CACHE` could indeed be set via the external environment, such as in your web server or in the console. Maybe that's the idea here? Whether or not there's an applicable place to point it out in docs or if this is just something that people understand as their Laravel knowledge increases, I'll leave that up to you guys :) But I tested by setting `SetEnv` in apache and it worked a treat.

**Comment by godruoyi** [comment]
hi @leonardchallis Yes, Laravel has a lot of magic and the documentation doesn't tell us how to use these imitations. We need to discover them ourselves. Regarding this issue, if you don't have any other updates, I think we can close it now.

**Comment by nunomaduro** [comment]
It is exceedingly rare for developers to customize the `APP_CONFIG_CACHE` environment variable. Therefore, we deliberately exclude this information from our documentation, as developers typically have no need to modify this value. We appreciate your suggestion, and if we receive more inquiries from customers about this topic, we will consider adding relevant information to our documentation.

**Comment by N30** [comment]
> It is exceedingly rare for developers to customize the `APP_CONFIG_CACHE` environment variable. Therefore, we deliberately exclude this information from our documentation, as developers typically have no need to modify this value. We appreciate your suggestion, and if we receive more inquiries from customers about this topic, we will consider adding relevant information to our documentation.

Been confused about this for years, I use it heavily as some of my applications and I keep checking the documentations to figure more about it but nothing has been added about it. One of my applications in particular is very dynamic, the files, database, session, configuration and everything related changes based on many factors and I really need to wrap my head around this. So just giving you a +1 on "more inquiries from customers on the topic" 

As for any personal suggestion in this case I have edited my bootstrap/app.php to after initial init of $app to put these environment variables dynamically and to prevent it from caching cross different domains (and in my case applications) I use this as my fire line of code, 

`putenv ("APP_CONFIG_CACHE=bootstrap/cache/".str_replace('.','_', $_SERVER['HTTP_HOST']  ) . "_config.php");`

but somehow in rare cases I still get funny things happening like apps trying to access DB with some defaults straight out of the vendor file. Sites trying to access with other site credentials, etc... thankfully this doesnt happen often enough on the production and usually goes unnoticed as is mostly in AJAX Livewire calls.
